From fc164acbbb700fd50ab9c04b47fc1b2687e9c0f4 Mon Sep 17 00:00:00 2001
From: "Bradley T. Hughes" <bradleythughes@fastmail.fm>
Date: Thu, 13 Oct 2016 07:38:38 +0000
Subject: [PATCH] build: add -DZLIB_CONST when building with --shared-zlib

Commit 782620f added the define only when building with the bundled
zlib. Using a shared zlib results in build breakage:

../src/inspector_agent.cc:179:16: error: assigning to 'Bytef *' (aka 'unsigned char *') from incompatible type
      'const uint8_t *' (aka 'const unsigned char *')
  strm.next_in = PROTOCOL_JSON + 3;
               ^ ~~~~~~~~~~~~~~~~~
1 error generated.
---
 node.gyp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/node.gyp b/node.gyp
index 272dc98..667c260 100644
--- a/node.gyp
+++ b/node.gyp
@@ -653,7 +653,8 @@
             [ 'node_shared_zlib=="false"', {
               'dependencies': [
                 'deps/zlib/zlib.gyp:zlib',
-              ]
+              ]}, {
+              'defines': [ 'ZLIB_CONST' ],
             }],
             [ 'node_shared_openssl=="false"', {
               'dependencies': [
